search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -149,46 +149,60 @@ class TemporalEntityHandler(
 }
 
 internal fun buildTemporalQuery(params: MultiValueMap<String, String>): TemporalQuery {
-    if (!params.containsKey("timerel") || !params.containsKey("time"))
-        throw BadRequestDataException("'timerel and 'time' request parameters are mandatory")
-
-    if (params.getFirst("timerel") == "between" && !params.containsKey("endTime"))
+    val timerelParam = params.getFirst("timerel")
+    val timeParam = params.getFirst("time")
+    val endTimeParam = params.getFirst("endTime")
+    val timeBucketParam = params.getFirst("timeBucket")
+    val aggregateParam = params.getFirst("aggregate")
+    val lastNParam = params.getFirst("lastN")
+    val attrsParam = params.getFirst("attrs")
+
+    if (timerelParam == "between" && endTimeParam == null)
         throw BadRequestDataException("'endTime' request parameter is mandatory if 'timerel' is 'between'")
 
-    val timerel = try {
-        TemporalQuery.Timerel.valueOf(params.getFirst("timerel")!!.toUpperCase())
-    } catch (e: IllegalArgumentException) {
-        throw BadRequestDataException("'timerel' is not valid, it should be one of 'before', 'between', or 'after'")
+    val (timerel, time) = if (timerelParam == null && timeParam == null) {
+        Pair(
+            TemporalQuery.Timerel.AFTER,
+            ("1970-01-01T00:00:00Z").parseTimeParameter("")
```

Comment:
does the spec say that?

Reply:
No but I guess I have to change the query for handling the absence of timerel value and time param if I don't give them default values. Should I do it ?

Reply:
I'll do it, since you' re off today...